accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,95 @@
+
+So, the way we will handle this is:
+
+- If `UseArtifactsOutput` is set to `true` in a `Directory.Build.props` file, then the new output format will be used, including for the intermediate output path
+- Otherwise, if `UseArtifactsOutput` is set to true later, either in the project file or in .NET SDK logic due to the `TargetFramework` being at least .NET 8, then the old output format will be used for the intermedate folder (so it will start with `obj\<Configuration>` by default), but the new `artifacts` format will be used for all other output.
+
+### Mixed output path formats in multi-targeted builds
+
+If a project is multi-targeted to `net7.0` and `net8.0` and doesn't specify the output path format, then different formats will be used for the output for the different target frameworks.  The .NET 7 build output will be in `bin\Debug\net7.0`, while the .NET 8 output will be in `artifacts\bin\debug_net8.0`.
+
+To prevent the output from one target framework to be globbed as part of the inputs to another target framework, both the `bin` and `artifacts` folder will need to be excluded from the default item globs (via the `DefaultItemExcludes` property).  This means that even projects that target .NET 7 or lower could be impacted by breaking changes if they currently have source files or other assets in an `artifacts` folder.
+
+Question: Could we have the outer build of a multi-targeted build query the inner builds to determine whether any of them target .NET 8 or higher, so we wouldn't have to have inconsistent output path formats?  Would this work with Visual Studio builds?
+
+### Solution or repo level output paths
+
+Many projects want to put all the output for the repo or the solution in a single folder.  We will support this with a new `RootArtifactsPath` property which can be set in a `Directory.Build.props file`.  If set, then we will use the following format for the output path:
+
+`<RootArtifactsPath>\<OutputType>\<ProjectName>\<Pivots>`
+
+If `RootArtifactsPath` is set, then it will not be necessary to separately set `UseArtifactsOutput`.
+
+We will consider updating commands such as `dotnet build` and `dotnet publish` so that when used on a `.sln` file, the `--output` parameter sets the `RootArtifactsPath`.
+
+## Considerations
+
+### Can't we use `bin\Debug`?
+
+Before .NET Core, .NET Framework projects generally put their output directly in `bin\Debug` or `bin\Release`.  It would be desirable if we could go back to putting the output in `bin\Debug`.  However, to do so we would need to do one of the following:
```

Comment:
> It would be desirable if we could go back to putting the output in `bin\Debug`.

Why? I'm not facetious here; knowing why we think it's desirable would help to understand if the proposed picking of two is indeed the right trade off. Personally, miss the simplicity of the older model, but I think the new system is better because it allows us to be simple and consistent.